api/hypershift/v1beta1/gcp.go

```diff
@@ -4,14 +4,55 @@ package v1beta1
 // See https://google.aip.dev/122 for GCP resource name standards.
 type GCPResourceReference struct {
-	// name is the name of the GCP resource
+	// name is the name of the GCP resource.
+	// Must conform to GCP resource naming standards: lowercase letters, numbers, and hyphens only.
+	// Must start with lowercase letter and end with lowercase letter or number, max 63 characters.
+	// Pattern: ^[a-z]([-a-z0-9]*[a-z0-9])?$ per GCP naming requirements.
+	// See https://cloud.google.com/compute/docs/naming-resources for details.
+	//
 	// +required
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:MaxLength=63
-	// +kubebuilder:validation:Pattern=`^[a-z]([a-z0-9]*(-[a-z0-9]+)*)?$`
+	// +kubebuilder:validation:Pattern=`^[a-z]([-a-z0-9]*[a-z0-9])?$`
 	Name string `json:"name"`
 }
 
+// GCPResourceLabel is a label to apply to GCP resources created for the cluster.
+// Labels are key-value pairs used for organizing and managing GCP resources.
+// See https://cloud.google.com/compute/docs/labeling-resources for GCP labeling guidance.
+type GCPResourceLabel struct {
+	// key is the key part of the label. A label key can have a maximum of 63 characters and cannot be empty.
+	// For Compute Engine resources (VMs, disks, networks created by CAPG), keys must follow RFC1035-style rules:
+	// - Start with a lowercase letter
+	// - Contain only lowercase letters, digits, or hyphens
+	// - Cannot end with a hyphen
+	// GCP reserves the 'goog' prefix for system labels.
+	// Note: Other GCP services may allow a broader character set (e.g., underscores); consult service-specific documentation.
+	// See https://cloud.google.com/compute/docs/labeling-resources for Compute Engine label requirements.
+	//
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=`^[a-z]([0-9a-z-]{0,61}[0-9a-z])?$`
+	// +kubebuilder:validation:XValidation:rule="!self.startsWith('goog')",message="Label keys starting with the reserved 'goog' prefix are not allowed"
+	Key string `json:"key"`
+
+	// value is the value part of the label. A label value can have a maximum of 63 characters.
+	// Empty values are allowed by GCP.
+	// For Compute Engine resources (VMs, disks, networks created by CAPG), values must follow RFC1035-style rules:
+	// - Contain only lowercase letters, digits, or hyphens
+	// - If non-empty, must start with a lowercase letter or digit
+	// - If non-empty, cannot end with a hyphen
+	// Note: Other GCP services may allow a broader character set (e.g., underscores); consult service-specific documentation.
+	// See https://cloud.google.com/compute/docs/labeling-resources for Compute Engine label requirements.
+	//
+	// +required
+	// +kubebuilder:validation:MinLength=0
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=`^$|^[0-9a-z]([0-9a-z-]{0,61}[0-9a-z])?$`
+	Value string `json:"value"`
```

Comment:
if empty string is a valid value, then the field cannot be required. If you want to make the field required AND allow empty values, then you should make it a pointer.